paddle/fluid/memory/detail/buddy_allocator.cc

```diff
@@ -71,7 +75,7 @@ void* BuddyAllocator::Alloc(size_t unaligned_size) {
 
   // refill the pool if failure
   if (it == pool_.end()) {
-    it = RefillPool();
+    it = RefillPool(size);
```

Comment:
Maybe better to change destructor as well. Change [https://github.com/PaddlePaddle/Paddle/pull/16295/files#diff-6b33aef9f13c1a2c3753d5d3fabf36adR45](url) to be `system_allocator_->Free(block, block->size(cache_), block->index(cache_));`.

Reply:
Not only constructor, I also found some similar places when call system_allcator->Free in this file. Please see my new change.